scripts/doc_checker.py

```diff
@@ -24,97 +40,242 @@
     *("PR08", "PR09", "PR10", "RT01", "RT04", "RT05", "SA02", "SA03"),
 }
 
-""" TEST CUSTOM CHECK
 MODIN_ERROR_CODES = {
-    "TE01": "'{parameter}' description should be '{should}', found - '{found}'"
+    "MD01": "'{parameter}' description should be '[type], default: [value]', found: '{found}'",
+    "MD02": "Spelling error in line: {line}, found: '{word}', reference: '{reference}'",
 }
 
 
-import inspect
+def get_optional_args(doc: Docstring) -> dict:
+    """
+    Get optional parameters for the object for which the docstring is checked.
 
+    Parameters
+    ----------
+    doc : numpydoc.validate.Doctring
+        Docstring handler.
 
-def get_default_args(func):
-    if not callable(func):
+    Returns
+    -------
+    dict
+        Dict with default arguments.
+    """
+    obj = doc.obj
+    if not callable(obj) or inspect.isclass(obj):
         return None
-    signature = inspect.signature(func)
+    signature = inspect.signature(obj)
     return {
         k: v.default
         for k, v in signature.parameters.items()
         if v.default is not inspect.Parameter.empty
     }
 
 
-def validate_modin_error(import_path):
-    from numpydoc.validate import Docstring
+def check_optional_args(doc: Docstring) -> list:
+    """
+    Check type description of optional arguments.
+
+    Parameters
+    ----------
+    doc : Docstring
+
+    Returns
+    -------
+    list
+        List of tuples with Modin error code and its description.
+    """
+    if not doc.doc_parameters:
+        return []
+    optional_args = get_optional_args(doc)
+    if not optional_args:
+        return []
 
-    doc = Docstring(import_path)
     errors = []
-    default_args = get_default_args(doc.obj)
-    if not default_args:
-        return None
-    for parameter in doc.doc_parameters:
-        if parameter in default_args:
-            type_line = doc.doc_parameters[parameter][0]
-            if "default " not in type_line:
-                after = type_line.split("default")[1]
-                found = "default" + after
-                should = "default [VALUE]"
-                errors.append(
-                    (
-                        "TE01",
-                        MODIN_ERROR_CODES["TE01"].format(
-                            parameter=parameter, should=should, found=found
-                        ),
-                    )
+    for parameter in optional_args:
+        type_line = doc.doc_parameters[parameter][0]
+        if "default: " not in type_line or "optional" in type_line:
```

Comment:
I think we can allow either `optional` or `default: VALUE` (but not both at the same time).

Example: a string parameter with default value of `None` meaning "do not consider this parameter". Declaring it as a string and saying that it defaults to `None` looks weird, so in this case I think we should name it just "optional".

Reply:
This makes sense but not each person knows about it. We can run into trouble with this. Let's do it so and then will see how it goes.

Reply:
So, let's do as @vnlitvinov suggested.